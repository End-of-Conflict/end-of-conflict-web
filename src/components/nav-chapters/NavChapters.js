/**
 * EoC NavChapters
 * @flow
 */
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './NavChapters.css';

type Props = {
  chapters: Array<{
    title: string,
    slug: string
  }>,
  language: string,
  toggleMainNav: () => *,
};

const NavChapters = (props: Props): null | React$Element<"li"> => {
  const { chapters, language, toggleMainNav } = props;
  const [open, setOpen] = useState(false);

  function toggleChapters() {
    setOpen(!open);
  }

  if (chapters.length === 0) {
    return null;
  }

  return (
    <li className={`NavChapters${open ? ' open' : ''}`}>
      <div
        className="toggle-chapters"
        onClick={toggleChapters}
        role="button"
        aria-controls="main-nav"
        aria-expanded={open}
      >
        Chapters
        <FontAwesomeIcon icon={`chevron-${open ? 'up' : 'down'}`} />
      </div>
      <ul>
        {
          chapters
            ? chapters.map((item, index) => {
                const {slug, title} = item;
                return (
                  <li key={index} onClick={toggleMainNav}>
                    <NavLink to={`/chapters/${language}/${slug}`}>
                      {title}
                    </NavLink>
                  </li>
                  )
              })
            : null
        }
      </ul>
    </li>
  );
}

NavChapters.defaultProps = {
  chapters: [{title: 'TLDR;', slug: 'tldr'}],
  language: 'en',
  toggleMainNav: (): null => null,
}

export default NavChapters;
