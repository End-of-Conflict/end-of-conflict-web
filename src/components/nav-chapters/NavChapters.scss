.NavChapters {
  ul {
    display: none;
  }

  &.open {
    ul {
      display: flex;
      flex-direction: column;
      align-items: flex-start;
      justify-content: center;
    }
  }

  .toggle-chapters {
    cursor: pointer;

    &:hover {
      text-decoration: underline;
    }

    .svg-inline--fa {
      margin-left: .5rem;

      &:hover {
        text-decoration: none;
      }
    }
  }
  
}