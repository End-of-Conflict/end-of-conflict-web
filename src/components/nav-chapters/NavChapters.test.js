import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import NavChapters from './NavChapters';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useRouteMatch: () => ({
    url: '/en/0',
    params: { chapter: 0 }
  }),
}));
configure({ adapter: new Adapter() });

describe('Renders', () => {
  test('NavChapters', () => {
    const wrapper = shallow(<NavChapters />);
    expect(wrapper).toMatchSnapshot();
  });
});
