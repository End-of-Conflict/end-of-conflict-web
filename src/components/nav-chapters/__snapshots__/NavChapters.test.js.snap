// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders NavChapters 1`] = `
<li
  className="NavChapters"
>
  <div
    aria-controls="main-nav"
    aria-expanded={false}
    className="toggle-chapters"
    onClick={[Function]}
    role="button"
  >
    Chapters
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-down"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <ul>
    <li
      key="0"
      onClick={[Function]}
    >
      <NavLink
        to="/chapters/en/tldr"
      >
        TLDR;
      </NavLink>
    </li>
  </ul>
</li>
`;
